compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -1584,7 +1584,8 @@ object Types extends TypeUtils {
         else if (tp.symbol.isAliasType) tp.underlying.underlyingClassRef(refinementOK)
         else NoType
       case tp: AppliedType =>
-        if (tp.tycon.isLambdaSub) NoType
+        if tp.isMatchAlias then tp.tryNormalize.underlyingClassRef(refinementOK)
```

Comment:
For this and 20258, what if it's a match alias that doesn't normalise, don't you want to cascade to the later cases?

Reply:
It might be incorrect but I had intentionally not cascaded to the later cases here. I don't think any of them would return something other than `NoType`.